pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -729,10 +727,16 @@ func (c *Controller) getProxyServiceInstancesByPod(pod *v1.Pod, service *v1.Serv
 		}
 	}
 
+	initEndpoint := c.newIstioEndpoint(pod, svc.Attributes)
```

Comment:
What do you think about calling this `newBasicIstioEndpoint` or `newEndpointTemplate` some thing like that and the other one as `newIstioEndpoint` or `newIstioEndpointFromTemplate` - I think that avoids some confusion that @howardjohn brought about?